### PR TITLE
OOP school library - Unit tests

### PR DESCRIPTION
**Pull request by elmar8287**
**I created this PR to check if our project was as requested.**

_In this project, we added unit tests to our codebase, specifically to our classes._

### Project requirements that are done:

> - Add tests files for the following classes: Book, Person, Student, Teacher, Classroom, Corrector, Rental ✔️ 
> - Write unit tests using Rspec. ✔️ 
 

<hr>

_Feel free to review my project and let me know if there are any essential changes 💪 ._

### CODE REVIEW

File: spec/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+require 'json'
+require_relative '../person'
+require_relative '../book'
+require_relative '../rental'
+require_relative '../teacher'
+require_relative '../student'
+
+describe 'Book' do
+  before :each do
```
Comment (SajjadAhmad14) [inline spec/book_spec.rb:9]: Please add tests for all the methods of the `Book` class.

File: spec/classroom_spec.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+require_relative '../classroom'
+
+describe Classroom do
+  before :each do
```
Comment (SajjadAhmad14) [inline spec/classroom_spec.rb]: Please add test for `add_student` method of `Classroom` class.

File: spec/person_spec.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+require_relative '../person'
+require_relative '../corrector'
+
+describe Person do
+  before(:each) do
```
Comment (SajjadAhmad14) [inline spec/person_spec.rb:5]: Please add tests for all public methods of the `Person` class.

File: spec/book_spec.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
+require 'json'
+require_relative '../person'
+require_relative '../book'
+require_relative '../rental'
+require_relative '../teacher'
+require_relative '../student'
+
```
Comment (SajjadAhmad14) [inline spec/book_spec.rb:7]: Please add `bundle install` and `rspec` commands in README so a non-technical user of the project can easily set it up locally.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi team, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
Git Flow is followed. :heavy_check_mark:
## To Improve 🙏

- [x] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity!

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: spec/classroom_spec.rb (hunk 2 above)
Comment (Donard97) [reply spec/classroom_spec.rb]: Thank you for letting us know. We did the test for the add_student method of the Classroom class!

File: spec/book_spec.rb (hunk 1 above)
Comment (Donard97) [reply spec/book_spec.rb:9]: Thank you for letting us know. We added all the tests for the Book class.

File: spec/person_spec.rb (hunk 3 above)
Comment (OLIPLICHE) [reply spec/person_spec.rb:5]: Thank you. We added the tests for all Public methods as you requested

File: spec/book_spec.rb (hunk 4 above)
Comment (OLIPLICHE) [reply spec/book_spec.rb:7]: Thank you. We have Updated the Readme file as you requested.

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi @Donard97 and @OLIPLICHE, 

Your project is complete!  There is nothing else to say other than... it's time to merge it! 
Congratulations! 🎉 

Great job✔✔

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by OLIPLICHE** [comment]
Thank you so much

**Comment by OLIPLICHE** [comment]
---------- Forwarded message ---------
From: Paka Olipliche Mav. ***@***.***>
Date: Wed, 5 Jan 2022 at 20:01
Subject: Re: Requirements
To: Elmar Abdulkarimov ***@***.***>


thank you

**Pull request merged**